hydromt_sfincs/sfincs.py

```diff
@@ -2600,6 +2604,208 @@ def setup_wind_forcing(self, timeseries=None, magnitude=None, direction=None):
         )
         self.set_forcing(da, name="wnd")
 
+    def setup_wave_forcing(
+        self,
+        geodataset: Union[str, Path, xr.Dataset] = None,
+        timeseries: list = None, #Union[str, Path, pd.DataFrame] = None, #This should now be a list with dataframes, or could we do that anotherway?
+        locations: Union[str, Path, gpd.GeoDataFrame] = None,
+        buffer: float = 5e3,
+        merge: bool = True,
+    ):
+        """Setup wave forcing.
+
+        Wave boundary conditions are read from a `geodataset` (geospatial point timeseries)
+        or a tabular `timeseries` dataframe. At least one of these must be provided.
+
+        The tabular timeseries data is combined with `locations` if provided,
+        or with existing 'snapwave_bnd' locations if previously set.
+
+        Adds model forcing layers:
+
+        * **wave_height** forcing: significant wave height time series [m]
+        * **wave_period** forcing: peak wave period time series [s]
+        * **wave_direction** forcing: mean wave direction time series [coming from degree wrt North]
+        * **wave_spreading** forcing: wave spreading time series [degree]            
+
+        Parameters
+        ----------
+        geodataset: str, Path, xr.Dataset, optional
+            Path, data source name, or xarray data object for geospatial point timeseries.
+        timeseries: str, Path, pd.DataFrame, optional
+            Path, data source name, or pandas data object for tabular timeseries.
+        locations: str, Path, gpd.GeoDataFrame, optional
+            Path, data source name, or geopandas object for bnd point locations.
+            It should contain a 'index' column matching the column names in `timeseries`.
+        buffer: float, optional
+            Buffer [m] around model wave boundary cells to select wave gauges,
+            by default 5 km.
+        merge : bool, optional
+            If True, merge with existing forcing data, by default True.
+
+        See Also
+        --------
+        set_forcing
+        """
+        gdf_locs, df_ts = None, None
+        
+        # If we wouldn't do any clipping, but just parse a geodataset directly:
+        # self.set_forcing(geodataset, name="snapwave", split_dataset=False)
```

Comment:
If this is something we want to do regularly, we could give the buffer an initial value of None and then only parse the geom and buffer to get_geodataset when "buffer" gets a numerical value?

Maybe also interesting for the setup_waterlevel function?

Reply:
done